### ISSUE TITLE
Controlling width of mermaid diagram for PDF generation

### ISSUE THREAD

**Issue by MatinF** [mgaitan/sphinxcontrib-mermaid#44]
I'm using this extension in my sphinx project, which works nicely for my html documentation. Here I set the output_format to `svg` and the charts render nicely.

However, when I render my latex and PDF files, some mermaid charts go out-of-page in terms of width. I would like this to be controlled so that they stay within the bounds. 

Is this something that should be done via the mermaid settings - or somewhere else?

I figured it might be related to the width defined of the underlying PNG images, but I've not been able to find a way to control this via the extension.

Similarly, I'm also struggling with how to avoid each mermaid diagram blocking text on the entire page in which they're added.

EDIT: I guess perhaps the issue here is that the mermaid diagrams are rendered as `svg` despite the use of `png` as the output format. This causes the mermaid diagrams to be included in my latex output as pdf files, rather than as `png` (which was what I expected). Note that I am able to render my mermaid diagrams as `png` when using the html build - so I'm not sure why it does not work for the latex build.

**Comment by MatinF** [comment]
I looked further into this and I can see part of my issue relates to the latest release being a bit outdated vs. master. I tried installing from master, but I am still unable to use `mermaid_pdfcrop` - I honestly do not understand what the syntax is supposed to look like for parsing this. Any hints or an example of how it's supposed to be used?

**Comment by ToddG** [comment]
@MatinF Did you get any further on this? I, too, want to generate PDFs and have the images embedded in them. However, I see no documentation for the config variables mentioned. And the link to `documentation` in the config section 404s. Sigh.

**Comment by MatinF** [comment]
Had a few notes from our README on how we used it, see below:

---------
Note that `1.8.5` is required rather than the latest version in order to render footnotes properly with the `sphinx_rtd_theme`.

Note that `sphinxcontrib-mermaid` commit `8679e1e` has been used in tests. Further, below modification should be made to the `mermaid.py` file once installed:
1. Add `shell=True` in both `Popen` calls: `p = Popen(mm_args, stdout=PIPE, stdin=PIPE, stderr=PIPE, )`
2. In `render_mm_latex` replace `pdf` by `png`


### Install mermaid CLI
```
npm install -g mermaid.cli 
```

Overall, however, we've found it difficult to work with in practice and for rendering. If I recall correctly, some modifications had to be made within the source code in order to make it work properly for e.g. rendering PDFs with images. I think there may be some github issues regarding this that I've commented on in the past, but I can't find them at this moment.

**Comment by ToddG** [comment]
Matin - I have this working and inserting images into PDF's is working as well. I did not have to make any modifications to the source:

* installed yarn into base os (ubuntu)
* created a yarn package in my project
* added mermaid-cli to the yarn package
* ran 'yarn' to download and install mermaid-cli
* ensured that mmdc options were specified in the sphinx conf.py

_conf.py_
```
"""Sphinx configuration."""
project = "(elided)"
author = "(elided)"
copyright = f"2020, {author}"
extensions = [
    "sphinx.ext.autodoc",
    "sphinx.ext.napoleon",
    "sphinx.ext.graphviz",
    "sphinx_autodoc_typehints",
    "sphinxcontrib.mermaid",
]
mermaid_cmd = "./node_modules/.bin/mmdc"
mermaid_output_format = "png"
mermaid_version = "8.8.3"```
```

* installed latexmk https://mg.readthedocs.io/latexmk.html
_ubuntu_
```
sudo apt install latexmk texlive-latex-base texlive-latex-extra texlive-latex-recommended
```

Then my noxfile build session is:

```
@nox.session(python=_versions)
def docs(session: Session) -> None:
    """Build the documentation."""
    session.run("poetry", "install", "--no-dev", external=True)
    install_with_constraints(
        session,
        "six",
        "sphinx",
        "sphinx-autodoc-typehints",
        "sphinxcontrib.mermaid",
    )
    session.run("sphinx-build", "-b", "html", "docs", "docs/_build/html")
    session.run("sphinx-build", "-b", "latex", "docs", "docs/_build/latex")
    session.cd("docs/_build/latex")
    session.run("make", external=True)
```
Today, I'll add these config options to my generic python seed app: https://github.com/ToddG/hypermodern-python-seed.

**Comment by markus-96** [comment]
regarding too big diagrams: you can avoid this issue by adding `-f` to the `mmdc` command. This will tell `mmdc` to fit the diagram on one page of the resulting .pdf. In my case, I added the following to my `conf.py`: `mermaid_cmd = "mmdc -f"` and since I am (having to) using Windows, I also added `mermaid_cmd_shell = "True"`.

TLDR: add `mermaid_cmd = "mmdc -f"` to `conf.py`
